main.go

```diff
@@ -374,7 +374,7 @@ func main() {
 		fmt.Println(err)
 		os.Exit(1)
 	}
-	if err := rootCmd.Execute(); err != nil {
+	if err := fang.Execute(context.Background(), rootCmd); err != nil {
 		_ = closer()
```

Comment:
Why not using a defer after setting up the logger?

Reply:
if my memory serves me, `os.Exit(1)` skips the `defer` calls 🤔

Reply:
Do you think there is a need to refactor then?

The question is for @caarlos0 @meowgorithm